gtsam/geometry/Cal3.cpp

```diff
@@ -0,0 +1,72 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   Cal3.cpp
+ * @brief  Common code for all calibration models.
+ * @author Frank Dellaert
+ */
+
+#include <gtsam/geometry/Cal3.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+
+namespace gtsam {
+using namespace std;
+
+/* ************************************************************************* */
+Cal3::Cal3(double fov, int w, int h)
+    : s_(0), u0_((double)w / 2.0), v0_((double)h / 2.0) {
+  double a = fov * M_PI / 360.0;  // fov/2 in radians
+  fx_ =
+      (double)w / (2.0 * tan(a));  //    old formula: fx_ = (double) w * tan(a);
+  fy_ = fx_;
+}
+
+/* ************************************************************************* */
+Cal3::Cal3(const std::string& path)
+    : fx_(320), fy_(320), s_(0), u0_(320), v0_(140) {
```

Comment:
I would prefer `double fx_ = 320.0f;` etc. in the .h (C++11 feature) instead of here at the .cpp. It's easier for the reader to see what are the defaults, and makes harder to be inconsistent between different ctors.

Reply:
I like your suggestion, but the default constructor uses `fx_ = 1`, `fy_ = 1` etc. Should I replace that with the defaults from this constuctor (from string path)?

Reply:
Hmm... good point. I think you should leave them like `double fx_ = 1.0;` etc. for consistency with the default ctor, then here in this ctor from a file, the defaults actually shouldn't matter since at the ctor should either: a) return with the new values from the file, or b) throw an exception if error loading... right?

Reply:
That's true. Let me update that really quick.